rinja_derive/src/generator.rs

```diff
@@ -2181,10 +2195,11 @@ pub(crate) fn is_cacheable(expr: &WithSpan<'_, Expr<'_>>) -> bool {
         Expr::Group(arg) => is_cacheable(arg),
         Expr::Tuple(args) => args.iter().all(is_cacheable),
         Expr::NamedArgument(_, expr) => is_cacheable(expr),
+        Expr::As(expr, _) => is_cacheable(expr),
+        Expr::Try(expr) => is_cacheable(expr),
         // We have too little information to tell if the expression is pure:
         Expr::Call(_, _) => false,
         Expr::RustMacro(_, _) => false,
-        Expr::Try(_) => false,
```

Comment:
If `e` is cacheable, then `e?` should be cacheable, too. What do you think?

Reply:
Hum... I don't see any reason not too. If tests are passing...